fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java

```diff
@@ -172,7 +174,12 @@ public CostEstimate visitPhysicalOlapScan(PhysicalOlapScanOperator node, Express
                     return adjustCostForMV(context);
                 }
             }
-            return CostEstimate.of(statistics.getComputeSize(), 0, 0);
+
+            // ScanOperator use a ChunkBuffer to implement async-io, which may takes significant memory for certain
+            // cases
+            SessionVariable variables = ConnectContext.get().getSessionVariable();
+            double memCost = (double) statistics.getAvgRowSize() * variables.getDegreeOfParallelism();
```

Comment:
why not * chunk_size?

Reply:
yeah, that can be better. but it would affect the query plan then i need to change a lot of test cases, which is i'm trying to avoid.